### PR TITLE
UI/NPC interaction

### PR DESCRIPTION
**Pull request by jidaprano**
- Added rudimentary pause menu
- Added modular settings page with ability to add new settings categories and settings within categories, supports dropdowns, sliders, and toggles
- Added NPC interactions that occur when getting close to NPCs

Video of how to modulate settings page: https://youtu.be/x2XAI6h32Xk

### CODE REVIEW

**Comment by smlendzian** [comment]
Unity won't let me test it until the merge conflict is resolved, so I'm going to add comments based on just looking at the code

File: Assets/Scripts/NPCSystem.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    private void Start()
+    {
+        npc = transform.parent;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(playerDetection && Input.GetKeyDown(KeyCode.E))
```
Comment (smlendzian) [inline Assets/Scripts/NPCSystem.cs:22]: (See previous comment) Should use the new Input system

File: Assets/Scripts/EscMenuController.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(Input.GetKeyDown(KeyCode.Escape))
+        {
+            if(pauseMenuPanel.activeSelf || settingsMenuPanel.activeSelf)
+            {
+                Resume();
```
[the remark this answers is not in the document]
Comment (jidaprano) [reply Assets/Scripts/EscMenuController.cs:25]: I agree wholeheartedly with this; however I think that may be out of scope of Timebox 1 with the time we have left until Wednesday, so I was going to leave this weak implementation until Timebox 2 when a better game state machine is implemented.

File: Assets/Scripts/MainMenu.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public void OnSettingsButton()
+    {
+        // Load settings screen
+        Debug.Log("Hit Settings Button");
+    }
+    public void OnQuitButton()
+    {
+        Application.Quit();
```
Comment (smlendzian) [inline Assets/Scripts/MainMenu.cs:22]: (See previous comment) Quit should be called from a game state manager

File: Assets/Scripts/EscMenuController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (smlendzian) [reply Assets/Scripts/EscMenuController.cs:25]: Cool

**Pull request merged**